airbyte-cdk/python/airbyte_cdk/utils/schema_inferrer.py

```diff
@@ -86,19 +86,34 @@ def accumulate(self, record: AirbyteRecordMessage) -> None:
         """Uses the input record to add to the inferred schemas maintained by this object"""
         self.stream_to_builder[record.stream].add_object(record.data)
 
+    def _null_type_in_any_of(self, node: InferredSchema) -> bool:
+        if "anyOf" in node:
+            return {"type": _NULL_TYPE} in node["anyOf"]
+        else:
+            return False
+
+    def _remove_type_from_any_of(self, node: InferredSchema) -> InferredSchema:
+        if "anyOf" in node:
+            node.pop("type", None)
+        return node
+
     def _clean(self, node: InferredSchema) -> InferredSchema:
         """
         Recursively cleans up a produced schema:
         - remove anyOf if one of them is just a null value
         - remove properties of type "null"
         """
+
         if isinstance(node, dict):
             if "anyOf" in node:
-                if len(node["anyOf"]) == 2 and {"type": _NULL_TYPE} in node["anyOf"]:
+                if len(node["anyOf"]) == 2 and self._null_type_in_any_of(node):
                     real_type = node["anyOf"][1] if node["anyOf"][0]["type"] == _NULL_TYPE else node["anyOf"][0]
                     node.update(real_type)
                     node["type"] = [node["type"], _NULL_TYPE]
                     node.pop("anyOf")
+                # populate `type` for `anyOf` if it's not present to pass all other checks
+                elif len(node["anyOf"]) == 2 and not self._null_type_in_any_of(node):
+                    node["type"] = [_NULL_TYPE]
```

Comment:
It feels strange to me to add the `null` type to the `anyOf` field, and then remove it at the end of this method.

It seems like it would make more sense to do one of the following:
1. Update the other checks to account for the `anyOf` case so that they allow the node to not have a `type` set, or
2. Add the `type: [object, null]` to the `anyOf` field and leave it there in the output schema

I'm not sure which approach is preferred. @girarda @maxi297 do either of you have an opinion here?

Reply:
Always adding `type: [object, null]` feels ok to me since we implicitly allow all columns to be nullable